consumer.go

```diff
@@ -335,7 +335,8 @@ func (c *Consumer) generateQuery() string {
 			sb.WriteString(` created_at >= CURRENT_TIMESTAMP - $3::interval AND`)
 			sb.WriteString(` created_at < CURRENT_TIMESTAMP AND`)
 		}
-		sb.WriteString(` (locked_until IS NULL OR locked_until < CURRENT_TIMESTAMP)`)
+		sb.WriteString(` (locked_until IS NULL OR locked_until < CURRENT_TIMESTAMP) AND`)
+		sb.WriteString(` delayed_until < CURRENT_TIMESTAMP`)
```

Comment:
Here can be some clash with the `HistoryLimit`.

We use the `HistoryLimit` because of partitioning.

@petans24 maybe the partitioning should have different condition? Is it possible?

The problematic scenario:

2 days old message (`created_at=NOW()-'2d'::interval`) with `delayed_until=NOW()` won't be consumed if the `HistoryLimit` is set to 1 day. It's a very edge case scenario, tho.

Reply:
I would just write to manual by BIG letters that delay interval cannot be longer than History Limit 😜
No reason to create overcomplex logic to handle this kind of edge situation (Maybe some runtime validation)

BTW: As I mentioned I'm from Prague so no problem to meet and discuss ;-)